### PR TITLE
CTCP-5575 Display incident location data

### PR DESCRIPTION
**Pull request by karahanozturk**
https://jira.tools.tax.service.gov.uk/browse/CTCP-5575

### CODE REVIEW

File: app/utils/IncidentP5Helper.scala [hunk 1]
```diff
@@ -66,17 +70,43 @@ class IncidentP5Helper(
...
 
+  private val selectLocationRow: String => Option[SummaryListRow] = qualifierOfIdentification =>
+    qualifierOfIdentification.toUpperCase match {
+      case "W" => coordinatesRow
+      case "U" => unLocodeRow
+      case "Z" => addressRow
+      case _ =>
+        logger.error(s"Unexpected qualifier of identification: $qualifierOfIdentification")
+        None
+    }
```
Comment (jackheslop96) [inline app/utils/IncidentP5Helper.scala:108]: Strictly speaking we're not really bothered about this logic. If one is defined the other two will be undefined. We can just safely add `coordinatesRow`, `unLocodeRow` and `addressRow` to our list of rows

File: app/utils/IncidentP5Helper.scala [hunk 2]
```diff
@@ -43,18 +47,18 @@ class IncidentP5Helper(
   )
 
   def countryRow: Option[SummaryListRow] = buildRowFromAnswer[String](
-    answer = Some("GB"), // TODO: Pull from incident data
+    answer = Some(data.Location.country),
```
Comment (jackheslop96) [inline app/utils/IncidentP5Helper.scala:50]: Is there any requirement to do a reference data lookup for this country?

File: app/utils/IncidentP5Helper.scala [hunk 3]
```diff
@@ -17,14 +17,18 @@
...
 import viewModels.sections.Section.StaticSection
 
 class IncidentP5Helper(
   data: IncidentType03
 )(implicit messages: Messages)
-    extends DeparturesP5MessageHelper {
+    extends DeparturesP5MessageHelper
+    with Logging {
+
+  val displayIndex = data.sequenceNumber + 1
```
Comment (jackheslop96) [inline app/utils/IncidentP5Helper.scala:31]: I can't see any tests for the IDs? I'm intrigued to see what `(data.sequenceNumber: String) + 1` will return

File: app/utils/IncidentP5Helper.scala (hunk 1 above)
Comment (kevincoatsworth) [reply app/utils/IncidentP5Helper.scala:108]: Yea just to hop on this, I'd had a quick look before I went on lunch - I was of the some opinion as Jack that we should be able to have them all in the list or rows as theoretically only one should ever be defined. I think the confusion has probably come from the IE182 postman data as more than one of these has been populated but in reality this shouldn't happen

File: app/utils/IncidentP5Helper.scala (hunk 3 above)
Comment (jackheslop96) [reply app/utils/IncidentP5Helper.scala:31]: @karahanozturk Fair. Sequence numbers start from 1, so the + 1 is not needed anyway. As long as we remember to unit test the IDs when actions get added it's all good.

File: app/utils/IncidentP5Helper.scala [hunk 4]
```diff
@@ -42,19 +51,27 @@ class IncidentP5Helper(
...
-    id = None,
-    call = None
-  )
+  def countryRow: Future[Option[SummaryListRow]] =
+    refDataService.getCountry(data.Location.country) map {
+      countryResponse =>
+        val countryToDisplay = countryResponse match {
+          case Right(country) => country.description
+          case Left(code)     => code
+        }
```
Comment (jackheslop96) [inline app/utils/IncidentP5Helper.scala:60]: Very minor but could use a fold here like `val countryToDisplay = countryResponse.fold[String](identity, _.description)`

**Pull request merged**